VERSION-RANGE-SPEC.rst

```diff
@@ -0,0 +1,989 @@
+======================================================
+vers: a mostly universal version range specifier
+======================================================
+
+This specification is a new syntax for dependency and vulnerable version ranges.
+
+
+Context
+--------
+
+Software package version ranges and version constraints are essential:
+
+- When resolving the dependencies of a package to express which subset of the
+  versions are supported. For instance a dependency or requirement statement
+  such as "I require package foo, version 2.0 or later versions" defines a
+  range of acceptable foo versions.
+
+- When stating that a known vulnerability or bug affects a range of package
+  versions. For instance a security advisory such as "vulnerability 123 affects
+  package bar, version 3.1 and version 4.2 but not version 5" defines a range of
+  vulnerable "bar" package versions.
+
+Version ranges can be replaced by a list enumerating all the versions of
+interest. But in practice, all the versions may not yet exist when defining an
+open version range such as "v2.0 or later".
+
+Therefore, a version range is a necessary, compact and practical way to
+reference multiple versions rather than listing all the versions.
+
+
+Problem
+--------
+
+Several version range notations exist and have evolved separately to serve the
+specific needs of each package ecosystem, vulnerability databases and tools.
+
+There is no (mostly) universal notation for version ranges and there is no
+universal way to compare two versions, even though the concepts that exist in
+most version range notations are similar.
+
+Each package type or ecosystem may define their own ranges notation and version
```

Comment:
It's even less consistent. In Python, there are multiple package managers, and not all package managers express versions the same way, even though they are the same ecosystem. https://python-poetry.org/docs/dependency-specification/#version-constraints

For Java, it's already noted that Gradle is similar to Maven, but there's also Ivy, which can load dependencies from Maven, but has its own version syntax. It also supports adding custom version constraints. https://ant.apache.org/ivy/history/latest-milestone/ivyfile/dependency.html

Reply:
Excellent point, to be handled in https://github.com/package-url/purl-spec/issues/328